clouddriver-kubernetes/src/main/groovy/com/netflix/spinnaker/clouddriver/kubernetes/v2/view/model/ManifestBasedModel.java

```diff
@@ -47,5 +50,18 @@ public String getProviderType() {
     return KubernetesCloudProvider.getID();
   }
 
+  public Moniker getMoniker() {
+    return NamerRegistry.lookup()
+        .withProvider(KubernetesCloudProvider.getID())
+        .withAccount(getAccountName())
+        .withResource(KubernetesManifest.class)
+        .deriveMoniker(getManifest());
+  }
+
+  public String getAccountName() {
+    return getKey().getAccount();
+  }
+
   abstract protected KubernetesManifest getManifest();
+  abstract protected Keys.InfrastructureCacheKey getKey();
```

Comment:
Looks like this new abstract method needs to be implemented in a couple of more classes.

Reply:
Ah whoops thanks